topic-operator/src/test/java/io/strimzi/operator/topic/v2/TopicControllerIT.java

```diff
@@ -421,6 +448,30 @@ private KafkaTopic createTopic(KafkaCluster kc, KafkaTopic kt) throws ExecutionE
         return waitUntil(created, readyIsTrueOrFalse());
     }
 
+    private void pauseTopic(String namespace, String topicName) {
+        var current = Crds.topicOperation(client).inNamespace(namespace).withName(topicName).get();
+        var paused = Crds.topicOperation(client).resource(new KafkaTopicBuilder(current)
+            .editMetadata()
+                .withAnnotations(ANNO_PAUSE_TRUE)
+            .endMetadata()
+            .build()).update();
+        LOGGER.info("Test paused KafkaTopic {} with resourceVersion {}",
+            paused.getMetadata().getName(), BatchingTopicController.resourceVersion(paused));
+        waitUntil(paused, pausedIsTrue());
+    }
+
+    private void updateTopicConfig(String namespace, String topicName, Map<String, Object> configs) {
```

Comment:
Can we not use the existing `modifyTopicAndAwait()` helper?

Reply:
Sure, but I need to adapt it to take into account paused state.